gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -0,0 +1,480 @@
+          .setProjectId("project id")
+          .setFeatureId("feature id")
+          .setLayerId("layer id")
+          .setFormId("form id")
+          .setUserId("user id")
+          .setResponseDeltas(
+              ImmutableList.of(
+                  ResponseDelta.builder()
+                      .setFieldId("field id")
+                      .setNewResponse(TextResponse.fromString("response for field id"))
+                      .build()))
+          .setClientTimestamp(new Date())
+          .build();
+
+  private static final Tile TEST_PENDING_TILE =
+      Tile.newBuilder()
+          .setId("id_1")
+          .setState(State.PENDING)
+          .setPath("some_path 1")
+          .setUrl("some_url 1")
+          .build();
+
+  private static final Tile TEST_DOWNLOADED_TILE =
+      Tile.newBuilder()
+          .setId("id_2")
+          .setState(State.DOWNLOADED)
+          .setPath("some_path 2")
+          .setUrl("some_url 2")
+          .build();
+
+  private static final Tile TEST_FAILED_TILE =
+      Tile.newBuilder()
+          .setId("id_3")
+          .setState(State.FAILED)
+          .setPath("some_path 3")
+          .setUrl("some_url 3")
+          .build();
+
+  private static final OfflineArea TEST_OFFLINE_AREA =
+      OfflineArea.newBuilder()
+          .setId("id_1")
+          .setBounds(LatLngBounds.builder().include(new LatLng(0.0, 0.0)).build())
+          .setState(OfflineArea.State.PENDING)
+          .build();
+
+  // This rule makes sure that Room executes all the database operations instantly.
+  @Rule public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+
+  @Inject LocalDataStore localDataStore;
+
+  private static void assertEqualsIgnoreId(
+      ObservationMutation expected, ObservationMutation actual) {
+    // TODO: Id is auto-assigned to ObservationMutation.
+    //  If we try to give it while inserting, then it causes problems. Improve this behavior.
+    //  So, copy the id from actual to expected and then compare the objects.
+    expected = expected.toBuilder().setId(actual.getId()).build();
+    assertThat(expected).isEqualTo(actual);
+  }
+
+  private static void assertObservation(ObservationMutation mutation, Observation observation) {
+    assertThat(mutation.getObservationId()).isEqualTo(observation.getId());
+    assertThat(mutation.getFeatureId()).isEqualTo(observation.getFeature().getId());
+    assertThat(TEST_USER).isEqualTo(observation.getCreated().getUser());
+    assertThat(TEST_FORM).isEqualTo(observation.getForm());
+    assertThat(TEST_PROJECT).isEqualTo(observation.getProject());
+    assertThat(TEST_USER).isEqualTo(observation.getLastModified().getUser());
+    MatcherAssert.assertThat(
+        ResponseMap.builder().applyDeltas(mutation.getResponseDeltas()).build(),
+        samePropertyValuesAs(observation.getResponses()));
+  }
+
+  private static void assertFeature(String featureId, Point point, Feature feature) {
```

Comment:
Same here - the name seems to be missing what is actually being asserted (i.e. or it seems to imply it's just checking that the supplied value is a feature)